nodeup/pkg/model/context.go

```diff
@@ -507,10 +508,15 @@ func (c *NodeupModelContext) NodeName() (string, error) {
 }
 
 // EvaluateHostnameOverride returns the hostname after replacing some well-known placeholders
-func EvaluateHostnameOverride(hostnameOverride string) (string, error) {
-	if hostnameOverride == "" || hostnameOverride == "@hostname" {
+func EvaluateHostnameOverride(hostnameOverride string, cloudProvider string) (string, error) {
+	if hostnameOverride == "@hostname" {
 		return "", nil
 	}
+
+	if hostnameOverride == "" && cloudProvider != "external" {
```

Comment:
I think also external cloudproviders generate the node names. But again, rather than generating hostnameOverride and testing on it, just test on cloudProvider.